astropy/table/table.py

```diff
@@ -2547,7 +2548,22 @@ def read(cls, *args, **kwargs):
         # import-time processing that collects available formats into an
         # RST table and inserts at the end of the docstring.  DO NOT REMOVE.
 
-        out = io_registry.read(cls, *args, **kwargs)
+        format = kwargs.pop('format', None)
+        if format and not args and not kwargs:
+            reader = io_registry.get_reader(format, cls)
+            reader_doc = ('Table.read() general documentation\n'
+                          '==================================\n')
```

Comment:
The headings don't look quite right - the top heading seems a bit 'stuck' to the Table.read call - maybe try with heading lines above too? (since I guess we don't need to be ReST compliant here?)